synapse/server.py

```diff
@@ -258,6 +256,9 @@ def __init__(
         self._module_web_resources: Dict[str, Resource] = {}
         self._module_web_resources_consumed = False
 
+        # This attribute is set by the free function `refresh_certificate`.
+        self.tls_server_context_factory: Optional[IOpenSSLContextFactory] = None
```

Comment:
Actually this might be redundant; I think mypy infers the type of instance attributes if they're set in the constructor like this? :shrug: